### PR TITLE
ref(ui) Replace more createReactClass usage

### PR DESCRIPTION
**Pull request by markstory**
Also update the MissingProjectMembership component to use more current icon assets and add missing margins.
Screen Shot 2019-08-23 at 12 45 27 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
-    return {
+    this.state = {
       loading: false,
       error: false,
       project,
     };
-  },
+  }
 
   joinTeam(team) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx]: This is bound in the event callback

File: src/sentry/static/sentry/app/views/settings/project/projectKeys/index.jsx [hunk 2]
```diff
@@ -316,7 +312,10 @@ export default class ProjectKeys extends AsyncView {
   }
 }
 
-const ClientKeyItemPanel = styled(({disabled, ...props}) => <Panel {...props} />)`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectKeys/index.jsx]: You could do:

```
const ClientKeyItemPanel = styled(({disabled: _disabled, ...props}) => <Panel {...props} />)`
```

File: src/sentry/static/sentry/app/views/settings/project/projectKeys/projectKeyDetails.jsx [hunk 3]
```diff
@@ -54,28 +53,29 @@ const RATE_LIMIT_FORMAT_MAP = new Map([
...
     };
-  },
+  }
 
   componentWillMount() {
     this.fetchData();
-  },
+  }
 
   fetchData() {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectKeys/projectKeyDetails.jsx]: We need to bind `this` due to L146

File: src/sentry/static/sentry/app/components/errorRobot.jsx [hunk 4]
```diff
@@ -24,19 +21,20 @@ const ErrorRobot = createReactClass({
     sampleIssueId: PropTypes.string,
 
     gradient: PropTypes.bool,
-  },
+  };
 
-  getInitialState() {
-    return {
+  constructor(props) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/errorRobot.jsx]: Should we use class props for state and remove the constructor?

File: src/sentry/static/sentry/app/components/events/interfaces/frame.jsx [hunk 5]
```diff
@@ -76,43 +73,43 @@ export const Frame = createReactClass({
...
-      isExpanded: false,
-      emptySourceNotation: false,
-    };
-  },
+  static defaultProps = {
+    isExpanded: false,
+    emptySourceNotation: false,
+  };
+
+  constructor(props) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/interfaces/frame.jsx]: Should we use class props for state and remove the constructor?

File: src/sentry/static/sentry/app/views/settings/components/settingsProjectItem.jsx [hunk 6]
```diff
@@ -1,30 +1,28 @@
...
+class ProjectItem extends React.Component {
+  static propTypes = {
     project: SentryTypes.Project,
     organization: SentryTypes.Organization,
-  },
+  };
 
-  getInitialState() {
-    return {
+  constructor(props) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/settingsProjectItem.jsx]: Should we use class props for state and remove the constructor?

File: src/sentry/static/sentry/app/views/settings/project/serviceHookSettingsForm.jsx [hunk 7]
```diff
@@ -13,20 +12,18 @@ import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
...
+  static propTypes = {
     orgId: PropTypes.string.isRequired,
     projectId: PropTypes.string.isRequired,
     hookId: PropTypes.string,
     initialData: PropTypes.object.isRequired,
-  },
+  };
 
-  onSubmitSuccess() {
+  onSubmitSuccess = () => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/serviceHookSettingsForm.jsx]: github's being weird, on L42 there's an `onCancel` method should that be removed?

File: src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx]: In all the places its used there is an existing `.bind()`. I clicked on the 'Join Team' button and it worked.

File: src/sentry/static/sentry/app/components/errorRobot.jsx (hunk 4 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/errorRobot.jsx]: The `sampleIssueId` state value is based on the prop value though.

File: src/sentry/static/sentry/app/views/settings/components/settingsProjectItem.jsx (hunk 6 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/components/settingsProjectItem.jsx]: How would the state derived from props work then?

File: src/sentry/static/sentry/app/components/errorRobot.jsx (hunk 4 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/errorRobot.jsx]: It'll work if you define it as

```javascript
    state = {
      error: false,
      loading: false,
      sampleIssueId: this.props.sampleIssueId,
    };
```

File: src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/projects/missingProjectMembership.jsx]: Oops yeah, this was just meant as confirmation

**Review by dashed — APPROVED** [review]
Looks good to me.

**Pull request merged**